massa-pos-exports/src/pos_final_state.rs

```diff
@@ -701,6 +720,52 @@ impl PoSFinalState {
         deferred_credits
     }
 
+    /// Gets the deferred credits for an address
+    pub fn get_address_deferred_credits(&self, address: &Address) -> BTreeMap<Slot, Amount> {
+        let db = self.db.read();
+
+        let mut deferred_credits = BTreeMap::new();
+
+        let start_key_buffer = DEFERRED_CREDITS_PREFIX.as_bytes().to_vec();
+
+        for (serialized_key, serialized_value) in db.iterator_cf(
+            STATE_CF,
+            MassaIteratorMode::From(&start_key_buffer, MassaDirection::Forward),
+        ) {
+            if !serialized_key.starts_with(DEFERRED_CREDITS_PREFIX.as_bytes()) {
+                break;
+            }
+            let (rest, slot) = self
+                .deferred_credits_deserializer
+                .slot_deserializer
+                .deserialize::<DeserializeError>(&serialized_key[DEFERRED_CREDITS_PREFIX.len()..])
+                .expect(DEFERRED_CREDITS_DESER_ERROR);
+
+            let (_, addr): (_, Address) = self
+                .deferred_credits_deserializer
+                .credit_deserializer
+                .address_deserializer
+                .deserialize::<DeserializeError>(rest)
+                .expect(DEFERRED_CREDITS_DESER_ERROR);
+
+            if &addr != address {
+                // TODO improve performance
+                continue;
```

Comment:
Note for a potential issue description: to do that, we might want instead to use a while loop, and query only the keys corresponding to the wanted address.
```rust
slot_serializer.serialize(cur_slot, &mut key)
addr_serializer.serialize(address, &mut key)
deferred_credit_prefix!(key)
```

In the case it's not found, go immediatly to the next existing slot key instead of iterating over all addresses.

Reply:
good, can you open a followup issue with description ?

Reply:
https://github.com/massalabs/massa/issues/4179